engine/frontend/src/component/CreateEnclaveView.js

```diff
@@ -27,38 +29,47 @@ export const CreateEnclaveView = ({packageId, enclave, args}) => {
-        const fetchLogs = async () => {
-          stream = await runStarlark(enclave.host, enclave.port, packageId, args, appData.jwtToken, appData.apiHost);
-          for await (const res of stream) {
-              const result = res["runResponseLine"]
-              if (result.case === "instruction") {
-                  setLogs(logs => [...logs, result.value.executableInstruction])
-              }
+    const readStreamData = (result) => {
+        if (result.case === "instruction") {
+            setLogs(logs => [...logs, result.value.executableInstruction])
+        }
 
-              if (result.case === "progressInfo" && result.value.currentStepInfo.length > 0) {
-                  console.log("progressinfo: ", result)
-                  setLogs(logs => [...logs, result.value.currentStepInfo[result.value.currentStepNumber]])
-              }
+        if (result.case === "progressInfo" && result.value.currentStepInfo.length > 0) {
+            if (result.value.currentStepInfo[result.value.currentStepNumber] !== undefined) {
+                setLogs(logs => [...logs, result.value.currentStepInfo[result.value.currentStepNumber]])
+            }
+        }
 
-              if (result.case === "instructionResult" && result.value.serializedInstructionResult) {
-                  if (result.value.serializedInstructionResult.includes(SERVICE_IS_ADDED)) {
-                      getServices(enclave)
-                  }
-                  setLogs(logs => [...logs, result.value.serializedInstructionResult])
-              }
+        if (result.case === "instructionResult" && result.value.serializedInstructionResult) {
+            if (result.value.serializedInstructionResult.includes(SERVICE_IS_ADDED)) {
+                getServices(enclave)
+            }
+            setLogs(logs => [...logs, result.value.serializedInstructionResult])
+        }
 
-              if (result.case === "error") {
-                const errorMessage = result.value.error.value.errorMessage;
-                setLogs(logs => [...logs, errorMessage])
-              }
-          }
-          setLoading(false);
+        if (result.case === "error") {
+            const errorMessage = result.value.error.value.errorMessage;
+            setLogs(logs => [...logs, errorMessage])
         }
+    }
 
-       fetchLogs();
+    useEffect(() => {
+        setLoading(true)
+        let stream;
+        const fetchLogs = async () => {
+            try {
+                stream = await runStarlark(enclave.host, enclave.port, packageId, args, appData.jwtToken, appData.apiHost);
+                for await (const res of stream) {
+                    const result = res["runResponseLine"]
+                    readStreamData(result)    
+                }
+            } catch (ex) {
+                console.log("Error occured while reading data from the enclave: ", enclave.name)
+            } finally {
+                setLoading(false)
+            }
+        }
+        fetchLogs();
```

Comment:
question for my understanding: Given that this `fetchLogs` is in `CreateEnclaveView`, I'm guessing these are enclave logs and not service logs. Is that accurate?

Reply:
yeah logs while creating enclave and services inside it.